handlers/getSignedUrl.js

```diff
@@ -0,0 +1,21 @@
+const { Storage } = require('@google-cloud/storage');
+const { getJsonFrom } = require('../common/api_adapter');
+const { bondBaseUrl, promiseHandler, Response, samBaseUrl, determineBondProvider } = require('../common/helpers');
+
+const getSignedUrl = promiseHandler(async (req) => {
+    const { bucket, object, dataObjectUri } = req.body || {};
+    const auth = req.headers.authorization;
+    const provider = dataObjectUri && determineBondProvider(dataObjectUri);
+    const credentials = provider ?
+        await getJsonFrom(`${bondBaseUrl()}/api/link/v1/${provider}/serviceaccount/key`, auth) :
```

Comment:
from line 8, isn't `provider` going to be a boolean?

Reply:
where's my compiler?

Reply:
Here we're using the common js pattern of relying on the fact that `undefined` is false-y.

Reply:
In js, `"a" && "b"` evaluates to `"b"`